pkg/executor/cms/cmscontroller.go

```diff
@@ -0,0 +1,180 @@
+				}
+				recyclePods(logger, funcs, ndm, gpm)
+			}
+
+		},
+	})
+	return store, controller
+}
+
+func getConfigmapRelatedFuncs(logger *zap.Logger, m *metav1.ObjectMeta, fissionClient *crd.FissionClient) ([]fv1.Function, error) {
+	funcList, err := fissionClient.Functions(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// In future a cache that populates at start and is updated on changes might be better solution
+	relatedFunctions := make([]fv1.Function, 0)
+	for _, f := range funcList.Items {
+		for _, cm := range f.Spec.ConfigMaps {
+			if (cm.Name == m.Name) && (cm.Namespace == m.Namespace) {
+				relatedFunctions = append(relatedFunctions, f)
+				break
+			}
+		}
+	}
+	return relatedFunctions, nil
+}
+
+func initSecretController(logger *zap.Logger, fissionClient *crd.FissionClient,
+	kubernetesClient *kubernetes.Clientset, ndm *nd.NewDeploy, gpm *gpm.GenericPoolManager) (cache.Store, cache.Controller) {
+	resyncPeriod := 30 * time.Second
+	listWatch := cache.NewListWatchFromClient(kubernetesClient.Core().RESTClient(), "secrets", metav1.NamespaceAll, fields.Everything())
+	store, controller := cache.NewInformer(listWatch, &apiv1.Secret{}, resyncPeriod, cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) {},
+		DeleteFunc: func(obj interface{}) {},
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			oldS := oldObj.(*apiv1.Secret)
+			newS := newObj.(*apiv1.Secret)
+			if oldS.ObjectMeta.ResourceVersion != newS.ObjectMeta.ResourceVersion {
+				if newS.ObjectMeta.Namespace != "kube-system" {
+					logger.Info("Secret changed",
+						zap.String("configmap_name", newS.ObjectMeta.Name),
+						zap.String("configmap_namespace", newS.ObjectMeta.Namespace))
+
+				}
+
+				funcs, err := getSecretRelatedFuncs(logger, &newS.ObjectMeta, fissionClient)
+				if err != nil {
+					logger.Error("Failed to get functions related to secret", zap.String("secret_name", newS.ObjectMeta.Name), zap.String("secret_namespace", newS.ObjectMeta.Namespace))
+				}
+				recyclePods(logger, funcs, ndm, gpm)
+			}
+
+		},
+	})
+	return store, controller
+
+}
+
+func getSecretRelatedFuncs(logger *zap.Logger, m *metav1.ObjectMeta, fissionClient *crd.FissionClient) ([]fv1.Function, error) {
+	funcList, err := fissionClient.Functions(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// In future a cache that populates at start and is updated on changes might be better solution
+	relatedFunctions := make([]fv1.Function, 0)
+	for _, f := range funcList.Items {
+		for _, secret := range f.Spec.Secrets {
+			if (secret.Name == m.Name) && (secret.Namespace == m.Namespace) {
+				relatedFunctions = append(relatedFunctions, f)
+				break
+			}
+		}
+	}
+	return relatedFunctions, nil
+}
+
+func recyclePods(logger *zap.Logger, funcs []fv1.Function, ndm *nd.NewDeploy, gpm *gpm.GenericPoolManager) {
```

Comment:
I don't really like the term `recycle` here since we are just trying to do the rolling update. Any thought? `rolloutNewPods`?

Reply:
I think Rollout is a bit confusing with Kubernetes's rollout and also in case of Pool Manager - we are simply deleting and not rolling out. I want to convey that we are taking old pods - fitting them with new configmaps/secrets and bringing them back to service. WDYT?